src/database/mariadb/mariadb-connection.ts

```diff
@@ -68,12 +63,13 @@ export class MariadbConnection implements DbConnection {
   async load(
     { collection, system, id }: Ref,
     deleted?: boolean,
+    forUpdate?: boolean,
   ): Promise<DbDocument> {
     const q = `SELECT *
       FROM ${this.conn.escapeId(collection)}
       WHERE system = ? AND id = ?
         ${deleted ? '' : 'AND deletedAt IS NULL'}
-      LIMIT 1`;
+      LIMIT 1 ${forUpdate ? 'FOR UPDATE' : ''}`;
```

Comment:
So do we add forUpdate to the request or...?

Reply:
No, that's passed through from the AppService if the record is being loaded for save or delete. This actually is what fixes the issue. Even on an update, MariaDB will delay this select until the other transaction finishes.